cmd/tempo-query/tempo/plugin.go

```diff
@@ -316,19 +347,45 @@ func (b *Backend) FindTraces(req *storage_v1.FindTracesRequest, stream storage_v
 	}
 
 	span.AddEvent(fmt.Sprintf("Found %d trace IDs", len(resp.TraceIDs)))
+	b.logger.Info("FindTraces: fetching traces", zap.Int("traceids", len(resp.TraceIDs)))
+
+	numWorkers := b.findTracesConcurrentRequests
+	jobs := make(chan job, len(resp.TraceIDs))
+	results := make(chan jobResult, len(resp.TraceIDs))
+	var wg sync.WaitGroup
+	// Start workers
+	for w := 0; w < numWorkers; w++ {
+		wg.Add(len(resp.TraceIDs))
```

Comment:
I think this is wrong, now the `wg` contains `numWorkers*numJobs`. Afaics it was correct before, it should contain the number of jobs.